lib/internal/worker.js

```diff
@@ -250,14 +251,18 @@ class Worker extends EventEmitter {
 
     this[kParentSideStdio] = { stdin, stdout, stderr };
 
-    const { port1, port2 } = new MessageChannel();
-    const transferList = [port2];
+    const mainThreadPortToWorker = createMainThreadPort(this.threadId);
+    const {
+      port1: publicPortToParent,
+      port2: publicPortToWorker,
+    } = new MessageChannel();
+    const transferList = [mainThreadPortToWorker, publicPortToWorker];
```

Comment:
Is it really ok to transfer the ownership of both ports but still keep a reference to `publicPortToParent` in `this[kPublicPort]`?
Doc tells "After transferring, they are not usable on the sending side of the channel
anymore"